kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,832 @@
+            y = int(new_bounds.bounds[1])
+            w = int(new_bounds.bounds[2])
+            h = int(new_bounds.bounds[3])
+            collides = self.collides_with_existent_stencil(x, y)
+            if collides == -1:
+                cliparea = StencilView(pos=(x, y), size=(w, h))
+                self.clip_rectangles.append(cliparea)
+                self.widget.add_widget(cliparea)
+                return len(self.clip_rectangles) - 1
+            else:
+                return collides
+        else:
+            return -2
+
+    def collides_with_existent_stencil(self, x, y):
+        idx = -1
+        for cliparea in self.clip_rectangles:
+            idx += 1
+            if self.contains(cliparea, x, y):
+                return idx
+        return -1
+
+    def draw_path(self, gc, path, transform, rgbFace=None):
+        points_line = []
+        polygons = path.to_polygons(transform, self.widget.width,
+                                    self.widget.height)
+        for polygon in polygons:
+            vertices = []
+            for x, y in polygon:
+                points_line += [float(x), float(y), ]
+            tess = Tesselator()
+            tess.add_contour(points_line)
+            if not tess.tesselate():
+                print("Tesselator didn't work :(")
+                return
+            newclip = self.handle_clip_rectangle(gc, x, y)
+            if newclip > -1:
+                self.draw_graphics(self.clip_rectangles[newclip], gc, tess,
+                                points_line, rgbFace)
+            else:
+                self.draw_graphics(self.widget, gc, tess, points_line, rgbFace)
+
+    def draw_graphics(self, widget, gc, polygons, points_line, rgbFace):
+        if isinstance(gc.line['dash_list'], tuple):
+            gc.line['dash_list'] = list(gc.line['dash_list'])
+        with widget.canvas:
+            if rgbFace is not None:
+                Color(*rgbFace)
+                for vertices, indices in polygons.meshes:
+                    Mesh(
+                        vertices=vertices,
+                        indices=indices,
+                        mode=str("triangle_fan")
+                    )
+            Color(*gc.get_rgb())
+            Line(points=points_line, width=int(gc.line['width'] / 2),
+                dash_length=gc.line['dash_length'],
+                dash_offset=gc.line['dash_offset'],
+                 dash_joint=gc.line['joint_style'],
+                 dash_list=gc.line['dash_list'])
+
+    def draw_image(self, gc, x, y, im):
+        bbox = gc.get_clip_rectangle()
+        if bbox is not None:
+            l, b, w, h = bbox.bounds
+        else:
+            l = 0
+            b = 0,
+            w = self.widget.width
+            h = self.widget.height
+        h, w = im.get_size_out()
+        rows, cols, image_str = im.as_rgba_str()
+        texture = Texture.create(size=(w, h))
+        texture.blit_buffer(image_str, colorfmt='rgba', bufferfmt='ubyte')
```

Comment:
One of the big issues is that instructions need to be cached and re-used rather than created every time. But we should leave that for much later.